source/containerimage/ocilayout.go

```diff
@@ -0,0 +1,195 @@
+package containerimage
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/reference"
+	"github.com/containerd/containerd/remotes"
+	"github.com/moby/buildkit/session"
+	sessioncontent "github.com/moby/buildkit/session/content"
+	"github.com/moby/buildkit/source"
+	digest "github.com/opencontainers/go-digest"
+	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// GetOCILayoutResolver gets a resolver to an OCI layout for a specified scope from the pool
+func GetOCILayoutResolver(layoutPath string, sm *session.Manager, sessionID string, g session.Group) *OCILayoutResolver {
+	r := &OCILayoutResolver{
+		path:      layoutPath,
+		sm:        sm,
+		sessionID: sessionID,
+		g:         g,
+	}
+	return r
+}
+
+type OCILayoutResolver struct {
+	remotes.Resolver
+	path      string
+	sm        *session.Manager
+	sessionID string
+	g         session.Group
+	is        images.Store
+	mode      source.ResolveMode
+}
+
+// WithImageStore returns new resolver that can also resolve from local images store
+func (r *OCILayoutResolver) WithImageStore(is images.Store, mode source.ResolveMode) *OCILayoutResolver {
```

Comment:
See the comment in `pull.go` . There is no `ImageStore` nor `Pusher` for OCILayout.

Reply:
Yeah `WithImageStore` removed.

Reply:
And `Pusher` too.